src/LogViewer/chunk-worker.js

```diff
@@ -127,6 +143,18 @@ const getParagraphClass = (lineNumber, { highlightStart, highlightEnd }) => {
     '';
 };
 
+// Tagged template function
+function escapeHtml(pieces) {
```

Comment:
In looking at this function, I notice a few things that can be cleaned with ES2015:

1. Use of `arguments`. This is almost never needed with ES2015 since we have rest parameters.
2. Variable extraction from parameters, which destructuring and rest parameters also solves.
3. Use of `forEach` to accumulate a value is an indicator that it can be simplified by using `Array#reduce`.

Putting these together, `escapeHtml` can be rewritten like so:

```js
const escapeHtml = ([result, ...strings], ...values) => values
  .reduce((result, value, index) => `${result}${escape(value)}${strings[index]}`, result);
```

A tagged template receives an array of strings and arguments for the values to be interpolated in between the strings. I renamed the variables to reflect that, so let's walk over the changes:

```js
function escapeHtml(pieces) {
  let result = pieces[0];
  const substitutions = [].slice.call(arguments, 1);

// becomes

const escapeHtml = ([result, ...strings],
```

The first argument to `escapeHtml` are the strings, so I destructured that into two variables: one for the initial `result`, and the other containing the rest of the `strings`. The remaining arguments are the `substitutions`, which I renamed to `values`.

```js
substitutions.forEach((elem, i) => {
  result += util.escape(elem) + pieces[i + 1];
});

return result;

// becomes

values.reduce((result, value, index) => `${result}${escape(value)}${strings[index]}`, result);
```

- Iterate over the values: `substitutions.forEach` vs. `values.reduce`
- Start with the current result: `result +=` vs. `.reduce(/**/, result)`
- Concatenate the previous result to the escaped value: `result += util.escape(elem)` vs. `${result}${escape(value)}`
- Concatenate the next piece: `+ pieces[i + 1]` vs. `${strings[index]}`

So the operations are the same, but a little more functional, succinct, and simplified. Also notice that `pieces[i + 1]` is different from `strings[index]` in that `pieces` didn't modify the original array, whereas `strings` was already destructured into an array with the `result` removed.

To recap, this is what the final `escapeHtml` function looks like:

```js
const escapeHtml = ([result, ...strings], ...values) => values
  .reduce((result, value, index) => `${result}${escape(value)}${strings[index]}`, result);
```